### PR TITLE
Verify Markdown files have current Table of Contents

### PR DESCRIPTION
**Pull request by ckadner**
**Description of your changes:**

* Update `tools/mdtoc.sh` script to remove colons from anchor links
* Add `check_mdtoc` target to `Makefile`
* Add `make check_mdtoc` to Travis/CI

**Environment tested:**

* Python Version: `3.7.5`
* Tekton Version: --
* Kubernetes Version: --
* Operating System: MacOS `10.15.5`

/cc @Tomcli

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,15 +37,19 @@ matrix:
...
       script:
         - make lint
+    - name: "Verify source files contain the license header"
+      language: bash
+      script:
+        - make check_license
+    - name: "Verify Markdown files have current table of contents"
+      language: bash
+      script:
+        - make check_mdtoc
```
Comment (Tomcli) [inline .travis.yml:52]: Can we group some these tests (e.g. license and mdtoc check) into one job before we start overwhelming the Travis CI quota?

Comment (ckadner) [reply .travis.yml:52]: What is the quota we have with Travis? Until we hit a limit I would like to keep them separate. Maybe there is a way to share common setup between Travis jobs but still report separate logical jobs?

Comment (Tomcli) [reply .travis.yml:52]: Right now it should be fine. For open source I think we are allowed to run up to 200 jobs in parallel per repo. So if we have more jobs, less commits/PR can be tested in parallel.

Comment (Tomcli) [reply .travis.yml:52]: This is something to put in mind as we creating more tests. For now, it LGTM

**Pull request merged**